### PR TITLE
[PREVIEW] PartitionKeyDelete: Adds public preview

### PR DESCRIPTION
**Pull request by Shaleen-Kalsi**
# PartitionKeyDelete public preview

## Description

This change puts the DeleteAllItemsByPartitionKeyStreamAsync API under the PREVIEW flag.

## Type of change

- [] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 1]
```diff
@@ -1525,7 +1525,7 @@ public abstract class Container
             string processorName,
             ChangeFeedStreamHandlerWithManualCheckpoint onChangesDelegate);
 
-#if INTERNAL
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:1528]: You are probably missing removing the method from ContainerInternal.

Also missing updating the Contract files, please see the test errors

File: Microsoft.Azure.Cosmos.Encryption.Custom/src/EncryptionContainer.cs [hunk 2]
```diff
@@ -1010,6 +1010,16 @@ internal sealed class EncryptionContainer : Container
             return this.ResponseFactory.CreateItemFeedResponse<T>(responseMessage);
         }
 
+#if PREVIEW
```
Comment (j82w) [inline Microsoft.Azure.Cosmos.Encryption.Custom/src/EncryptionContainer.cs:1013]: This is the encryption project. It builds using both a project reference and the public nuget. You need to use the following flag to get it to build correctly. The entire encryption project is in preview. 

Example: https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2935
```suggestion
#if SDKPROJECTREF
```

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs (hunk 1 above)
Comment (Shaleen-Kalsi) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:1528]: fixed

File: Microsoft.Azure.Cosmos.Encryption.Custom/src/EncryptionContainer.cs (hunk 2 above)
Comment (Shaleen-Kalsi) [reply Microsoft.Azure.Cosmos.Encryption.Custom/src/EncryptionContainer.cs]: made the fix

**Review by kirankumarkolli — APPROVED** [review]

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**